journal.py

```diff
@@ -0,0 +1,175 @@
+# _*_ coding: utf-8 _*_
+from __future__ import unicode_literals
+import os
+from pyramid.config import Configurator
+from pyramid.view import view_config
+from waitress import serve
+import sqlalchemy as sa
+from sqlalchemy.ext.declarative import declarative_base
+import datetime
+from sqlalchemy.orm import scoped_session, sessionmaker
+from zope.sqlalchemy import ZopeTransactionExtension
+from pyramid.httpexceptions import HTTPFound
+from sqlalchemy.exc import DBAPIError
+from pyramid.authentication import AuthTktAuthenticationPolicy
+from pyramid.authorization import ACLAuthorizationPolicy
+from cryptacular.bcrypt import BCRYPTPasswordManager
+from pyramid.security import remember, forget
+
+DBSession = scoped_session(
+    sessionmaker(extension=ZopeTransactionExtension())
+)
+Base = declarative_base()
+
+
+class Entry(Base):
+    __tablename__ = 'entries'
+    id = sa.Column(sa.Integer, primary_key=True, autoincrement=True)
+    title = sa.Column(sa.Unicode(128), nullable=False)
+    text = sa.Column(sa.UnicodeText, nullable=False)
+    timestamp = sa.Column(
+        sa.DateTime, nullable=False, default=datetime.datetime.utcnow)
+
+    @classmethod
+    def write(cls, title=None, text=None, session=None):
+        if session is None:
+            session = DBSession
+        new_entry = cls(title=title, text=text)
+        session.add(new_entry)
+        return new_entry
+
+    @classmethod
+    def all(cls, session=None):
+        if session is None:
+            session = DBSession
+        return session.query(cls).order_by(cls.timestamp.desc()).all()
+
+    def __repr__(self):
+        return "<Entry(title='%s', timestamp='%s')>" % (
+            self.title, self.timestamp
+        )
+
+DATABASE_URL = os.environ.get(
+    'DATABASE_URL',
+    "postgresql://Scott@localhost:5432/learning-journal"
+)
+
+
+def init_db():
+    engine = sa.create_engine(DATABASE_URL, echo=True)
+    Base.metadata.create_all(engine)
+
+
+@view_config(route_name='home', renderer='templates/list.jinja2')
+def list_view(request):
+    entries = Entry.all()
+    return {'entries': entries}
+
+
+@view_config(route_name='new-entry', renderer='templates/new-entry.jinja2')
+def new_entry(request):
+    return {}
+
+
+@view_config(route_name='detail', renderer='templates/detail.jinja2')
+def post_detail(request):
+    entries = Entry.all()
+    return {'entries': entries}
```

Comment:
I'm not sure you want to show *all* the entries in a view meant to display the details of one entry.  You might think about extending the API of your entry class to provide a method that gets one entry using some identifying information.